lib/vmdb/loggers/azure_logger.rb

```diff
@@ -0,0 +1,22 @@
+module Vmdb::Loggers
+  class AzureLogger < VMDBLogger
+    def initialize(*loggers)
+      super
+
+      @datetime_format = '%Y-%m-%d %H:%M:%S'
+
+      @formatter = proc do |severity, datetime, _progname, msg|
+        msg = msg.sub(/Bearer(.*?)\"/, 'Bearer [FILTERED] "')
+        msg = msg.sub(/SharedKey(.*?)\"/, 'SharedKey [FILTERED] "')
+        msg = msg.sub(/client_secret=(.*?)&/, "client_secret=[FILTERED]&")
+        "[#{datetime}] - #{severity} -- : #{msg}\n"
+      end
+    end
+
+    def <<(msg)
+      msg = msg.strip
+      log(level, msg)
+      msg.size
+    end
```

Comment:
Why is this necessary?  A VMDBLogger  is a Logger, which already has the `<<` method defined as you have done here.

Reply:
Not sure why, but it won't work without it. I assume it's a rest-client issue.

Reply:
What kind of error?  I just tried << on a VMDBLogger and it works.

Reply:
Ohhh, I see... `<<` "works", but doesn't write out the timestamp, which is the problem in the first place.  I'm wondering if this should just be implemented on the underlying VMDBLogger class instead.  @NickLaMuro @bdunne thoughts?

Reply:
> I'm wondering if this should just be implemented on the underlying VMDBLogger class instead.

I don't know, seems like a bad idea.  The whole point of `Logger#<<` is to print without formatting:

https://github.com/ruby/ruby/blob/trunk/lib/logger.rb#L477-L483

I think it is stupid for `rest-client` to log via `<<`, so I think doing this is a targeted class makes more sense.  We only want this with loggers where this is a problem (i.e., ones that have to hook into `rest-client`), and not for every logger in MIQ, because the original behavior of `#<<` might be desired in some other circumstance.

Reply:
> I think it is stupid for `rest-client` to log via `<<`

:+1: Has anyone looked into potentially fixing this?

Reply:
@bdunne Yes, we have. Take a look at the issue tracker for rest-client. Unfortunately, the project is in poor health.

Reply:
I have been looking into replacing `rest-client` where we can (since a couple of dependencies for it our ones we own).  That said, it is more a labor of love, and not something on the "official agenda".

Reply:
@NickLaMuro I'm desperately trying to get azure-armrest switched over to Excon, but it's a lot of work. I should probably make a separate port where I don't try to refactor anything as I go, and just do a straight port to get it working.

Reply:
@NickLaMuro I actually thinking moving off of rest-client should start getting prioritized a bit higher the longer that rest-client languishes. I think Excon and Faraday are the two best options at this point.

Reply:
I was just going to use `net/http` where the additions of provided by other libs aren't really necessary.

:neckbeard: